src/main/java/lab2/Lab2.java

```diff
@@ -0,0 +1,199 @@
+classes:
+top-level windows - JFrame, JDialog
+user interface elements - JLabel, JButton, JTextField, JPanel
+containers and layout managers
+        */
+import javax.swing.*;
+/*package: java.awt
+purpose: used to create GUI and graphics, mostly common in older GUI.
+kinds of Classes:
+component classes - Frame, Button, Label
+graphics classes - Graphics, Color, Font, Image
+layout managers - FlowLayout, BorderLayout, GridLayout
+event handling - ActionEvent, MouseEvent
+ */
+import java.awt.*;
+/*package: java.io.IOException
+part of the Java Image I/O API, used to read, write, and manipulate image files.
+kinds of Classes:
+imageIO – provides static methods like read() and write() for handling image files.
+imageReader, ImageWriter – for low-level image format operations.
+imageTypeSpecifier, ImageInputStream, ImageOutputStream
+*/
+import java.io.IOException;
+/*package: java.io.InputStream
+purpose: provides classes for system input and output through data streams, serialization, and file system access.
+ Useful for reading/writing bytes, characters, and objects.
+classes: Windows (JFrame), buttons (JButton), labels (JLabel), panels (JPanel), and many others.
+*/
+import java.io.InputStream;
+/*package: java.net
+purpose: used for networking, such as working with URLs, sockets, and internet addresses.
+classes:
+URI, URL – represent web addresses
+Socket and ServerSocket – for network communication
+InetAddress and URLConnection
+ */
+import java.net.URI;
+/*
+kinds of Classes:
+HttpClient – sends HTTP requests
+HttpRequest – builds HTTP requests
+HttpResponse – handles responses
+HttpRequest.BodyPublisher, HttpResponse.BodyHandler - control body types
+
+ */
+import java.net.http.HttpClient;
+/*
+purpose: represents the HTTP client. It is used to send HTTP requests and receive responses.
+constructor: HttpClient.newHttpClient() to create an instance (factory method)
+classes:
+.send(request, handler) -	Sends a synchronous request
+.sendAsync(request, handler)- Sends an asynchronous request
+ */
+
+import java.net.http.HttpRequest;
+
+/*
+purpose: represents an HTTP response from a server. It contains status code, headers, and body.
+classes:
+
+HttpClient - Main class used to send requests and receive responses. Can be configured with options like redirect policy, proxy, and SSL.
+HttpRequest	- Represents an HTTP request. Built using a builder pattern (HttpRequest.newBuilder()).
+HttpResponse<T>	- Represents an HTTP response, with generic type T for the response body (e.g., String, InputStream).
+HttpRequest.BodyPublishers - Contains static methods to create request bodies (e.g., ofString(), ofFile()).
+HttpResponse.BodyHandlers -	Contains static methods to process response bodies (e.g., ofString(), ofInputStream()).
+WebSocket - Provides support for WebSocket communication (real-time, full-duplex).
+HttpHeaders - Represents HTTP headers (used in requests and responses).
+HttpTimeoutException - Exception thrown when a request times out.
```

Comment:
You've covered a lot of the types used in the code here, but not all of them, and some of the types you discuss here are not actually used in this code.